langworld_db_data/featureprofiletools/value_renamer.py

```diff
@@ -0,0 +1,113 @@
+from pathlib import Path
+
+from langworld_db_data.constants.paths import FEATURE_PROFILES_DIR, INVENTORIES_DIR
+from langworld_db_data.filetools.csv_xls import read_dicts_from_csv, write_csv
+
+
+class ValueRenamerError(Exception):
+    pass
+
+
+class ValueRenamer:
+    def __init__(
+        self,
+        input_feature_profiles_dir: Path = FEATURE_PROFILES_DIR,
+        input_inventories_dir: Path = INVENTORIES_DIR,
+        output_feature_profiles_dir: Path = FEATURE_PROFILES_DIR,
+        output_inventories_dir: Path = INVENTORIES_DIR,
+    ) -> None:
+        self.input_feature_profiles_dir = input_feature_profiles_dir
+        self.input_inventories_dir = input_inventories_dir
+        self.output_feature_profiles_dir = output_feature_profiles_dir
+        self.output_inventories_dir = output_inventories_dir
+
+    def rename_value_in_profiles_and_inventories(
+        self,
+        id_of_value_to_rename: str,
+        new_value_name: str,
+    ) -> None:
+        """
+        Renames value with given ID in all feature profiles and inventory of listed values.
+        Also works with elementary values in multiselect features.
+        """
+        files = list(self.input_feature_profiles_dir.glob("*.csv"))
+        for file in files:
+            print(f"Processing {file.name}")
+            self._update_one_feature_profile(
+                id_of_value_to_rename=id_of_value_to_rename,
+                new_value_name=new_value_name,
+                input_file=file,
+                output_dir=self.output_feature_profiles_dir,
+            )
+
+        if not self.output_inventories_dir.exists():
+            self.output_inventories_dir.mkdir(parents=True, exist_ok=True)
+        self._update_features_listed_values(
+            id_of_value_to_rename=id_of_value_to_rename,
+            new_value_name=new_value_name,
+            input_file=self.input_inventories_dir / "features_listed_values.csv",
+            output_file=self.output_inventories_dir / "features_listed_values.csv",
+        )
+
+    @staticmethod
+    def _update_one_feature_profile(
+        id_of_value_to_rename: str,
+        new_value_name: str,
+        input_file: Path,
+        output_dir: Path,
+    ) -> None:
+
+        number_of_replacements = 0
+        data_from_file = read_dicts_from_csv(input_file)
+        data_to_write = []
+        for line in data_from_file:
+            if id_of_value_to_rename not in line["value_id"]:
+                data_to_write.append(line)
+                continue
+            line_to_write = line.copy()
+            # After this clause, only lines with the target value are considered (they may contain other values too)
+            if "&" in line["value_id"]:
+                print("Found match in combined value in " + input_file.name)
```

Comment:
```suggestion
                print(f"Found match in combined value in {input_file.name}")
```
this is just considered better coding practice AFAIK